### PR TITLE
Manage Holidays: Create Holiday

### PR DESCRIPTION
**Pull request by DivineWisdom**
## Description
Manage Holiday: Create Holiday

## Related issues and discussion
#154

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/organization/holidays/create-holiday/create-holiday.component.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                            Name <strong>cannot</strong> begin with a special character or number
+                        </mat-error>
+                    </mat-form-field>
+
+                    <mat-form-field>
+                        <mat-label>From date</mat-label>
+                        <input matInput [min]="minDate" [max]="maxDate" formControlName="openingDate" required [matDatepicker]="openedOnDatePicker">
+                        <mat-datepicker-toggle matSuffix [for]="openedOnDatePicker"></mat-datepicker-toggle>
+                        <mat-datepicker #openedOnDatePicker></mat-datepicker>
+                        <mat-error *ngIf="officeForm.controls.openingDate.hasError('required')">
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.html]: It should be holiday form.

File: src/app/organization/holidays/create-holiday/create-holiday.component.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                            From Date is <strong>required</strong>
+                        </mat-error>
+                    </mat-form-field>
+
+                    <mat-form-field>
+                        <mat-label>To date</mat-label>
+                        <input matInput [min]="minDate" [max]="maxDate" formControlName="openingDate" required [matDatepicker]="openedOnDatePicker">
+                        <mat-datepicker-toggle matSuffix [for]="openedOnDatePicker"></mat-datepicker-toggle>
+                        <mat-datepicker #openedOnDatePicker></mat-datepicker>
+                        <mat-error *ngIf="officeForm.controls.openingDate.hasError('required')">
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.html]: It should be holiday form.

File: src/app/organization/holidays/create-holiday/create-holiday.component.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        <mat-datepicker-toggle matSuffix [for]="openedOnDatePicker"></mat-datepicker-toggle>
+                        <mat-datepicker #openedOnDatePicker></mat-datepicker>
+                    </mat-form-field>
+
+                    <mat-form-field>
+                        <mat-label>Repayment Scheduled to</mat-label>
+                        <input matInput [min]="minDate" [max]="maxDate" formControlName="openingDate" required [matDatepicker]="openedOnDatePicker">
+                        <mat-datepicker-toggle matSuffix [for]="openedOnDatePicker"></mat-datepicker-toggle>
+                        <mat-datepicker #openedOnDatePicker></mat-datepicker>
+                        <mat-error *ngIf="officeForm.controls.openingDate.hasError('required')">
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.html]: It should be holiday form.

File: src/app/organization/holidays/create-holiday/create-holiday.component.html [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                            Description is <strong>required</strong>
+                        </mat-error>
+                    </mat-form-field>
+                </div>
+
+            </mat-card-content>
+
+            <mat-card-actions fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center" fxLayoutGap="5px">
+                <button type="button" mat-raised-button [routerLink]="['../']">Cancel</button>
+                <button mat-raised-button color="primary" [disabled]="!employeeForm.valid">Submit</button>
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.html]: it should be holiday form.

File: src/app/organization/holidays/create-holiday/create-holiday.component.html [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            </mat-card-actions>
+
+        </form>
+
+    </mat-card>
+
+</div>
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.html]: add empty line here.

File: src/app/organization/holidays/create-holiday/create-holiday.component.scss [hunk 6]
```diff
@@ -0,0 +1,7 @@
+.container {
+    max-width: 37rem;
+}
+
+.loan-officer {
+    padding: 17.5px 0 17.5px 0;
+}
```
Comment (spider0061) [inline src/app/organization/holidays/create-holiday/create-holiday.component.scss]: add empty line here.

File: src/environments/environment.ts [hunk 7]
```diff
@@ -9,8 +9,8 @@ import env from './.env';
 export const environment = {
   production: false,
   version: env.npm_package_version + '-dev',
-  fineractPlatformTenantId: 'default',  // For connecting to server running elsewhere update the tenant identifier
-  baseApiUrl: 'https://demo.openmf.org',  // For connecting to server running elsewhere update the base API URL
+  fineractPlatformTenantId: 'staging',  // For connecting to server running elsewhere update the tenant identifier
```
Comment (spider0061) [inline src/environments/environment.ts]: don't push environment.ts in your PR.

Comment (DivineWisdom) [reply src/environments/environment.ts]: > don't push environment.ts in your PR.

Okay, I will make the necessary adjustments.

**Comment by DivineWisdom** [comment]
@spider0061 I have made the changes, please review

File: src/environments/environment.ts (hunk 7 above)
Comment (spider0061) [reply src/environments/environment.ts:12]: @DivineWisdom I think you are using git add . to add your changes. I suggest you to use git add address of the source folder in which you have made changes and then the unnecessary changes will not be pushed. example here you can use git add src/app/organization as you are making changes to the organization folder.

File: src/app/organization/employees/create-employee/create-employee.component.scss [hunk 8]
```diff
@@ -1,7 +1,7 @@
...
-}
-
+    max-width: 37rem;
+  }
+  
 .loan-officer {
-  padding: 17.5px 0 17.5px 0;
-}
+    padding: 17.5px 0 17.5px 0;
+  }
```
Comment (muskankhedia) [inline src/app/organization/employees/create-employee/create-employee.component.scss:7]: @DivineWisdom Remove these unnecessary changes. The linting and indentation are kept in mind and well maintained while writing the code. Please avoid making such changes in the files.

File: src/app/organization/holidays/holidays.component.html [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+            <tr mat-row *matRowDef="let row; columns: displayedColumns;" [routerLink]="[row.id]" class="select-row"></tr>
+
+        </table>
+        <mat-paginator [pageSizeOptions]="[10, 25, 50, 100]" showFirstLastButtons></mat-paginator>
+
+
+
+    </div>
+
+</div>
```
Comment (muskankhedia) [inline src/app/organization/holidays/holidays.component.html:68]: Do add a line at the end of the file.

File: src/app/organization/holidays/holidays.component.scss [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    }
+}
+
+.true {
+    color: #32cd32;
+}
+
+.false {
+    color: #f44366;
+}
```
Comment (muskankhedia) [inline src/app/organization/holidays/holidays.component.scss:14]: Do add a line at the end of the file.

File: src/environments/environment.prod.ts [hunk 11]
```diff
@@ -4,8 +4,8 @@ import env from './.env';
 export const environment = {
   production: true,
   version: env.npm_package_version,
-  fineractPlatformTenantId: 'staging',  // For connecting to server running elsewhere update the tenant identifier
-  baseApiUrl: 'https://staging.openmf.org',  // For connecting to server running elsewhere update the base API URL
+  fineractPlatformTenantId: 'default',  // For connecting to server running elsewhere update the tenant identifier
+  baseApiUrl: 'https://demo.openmf.org',  // For connecting to server running elsewhere update the base API URL
```
Comment (muskankhedia) [inline src/environments/environment.prod.ts:8]: Avoid making any changes to the environment.prod.ts file or any env based files.

**Comment by karantakalkar** [comment]
@DivineWisdom can you resolve merge conflicts and edit PR as per suggestions please?, also let us know if you will not be able to update the PR for now.

**Comment by DivineWisdom** [comment]
@karantakalkar sorry for my inactivity lately was currently having health issues. And yes I will resolve the merge conflicts and update the PR

**Comment by karantakalkar** [comment]
@DivineWisdom alright then

File: src/app/organization/employees/create-employee/create-employee.component.scss (hunk 8 above)
Comment (DivineWisdom) [reply src/app/organization/employees/create-employee/create-employee.component.scss:7]: @muskankhedia  sorry for the disorder but due to the way my code editor (VS Code) was formatted, it automatically changed the structure of the original code. I have addressed that issue though.

Comment (muskankhedia) [reply src/app/organization/employees/create-employee/create-employee.component.scss:7]: Okay, please revert it.

**Comment by muskankhedia** [comment]
@DivineWisdom try using `ng lint` and fixing all the errors before making the commit. That will help you to pass the TravisCI check.

**Comment by DivineWisdom** [comment]
> @DivineWisdom try using `ng lint` and fixing all the errors before making the commit. That will help you to pass the TravisCI check.

Alright, thanks

**Comment by karantakalkar** [comment]
@DivineWisdom there are a few contributors who would like to work on this issue, let me know if you won't get time to update so that others can work on it.
